### PR TITLE
Update version of react-native-reanimated to 2.4.0

### PR DESCRIPTION
**Pull request by yedidyak**
## Description
Update version of react-native-reanimated (2.4.0)

## Changelog
Update internally versions of react-native-reanimated and react-native-gesture-handler

### CODE REVIEW

**Comment by M-i-k-e-l** [comment]
For future reference a few workarounds were aded:
1. `onMomentumScrollEnd={() => {}}` workaround for `useAnimatedScrollHandler.onMomentumEnd` not being called (Android) (https://github.com/software-mansion/react-native-reanimated/issues/2735)
2. `moduleNameMapper` in `package.json` see this: https://github.com/software-mansion/react-native-reanimated/issues/1196
3. `interpolateColor` mock see this https://github.com/software-mansion/react-native-reanimated/issues/2749

File: src/components/tabController/PageCarousel.tsx [hunk 1]
```diff
@@ -103,6 +101,7 @@ function PageCarousel({...props}) {
       scrollEventThrottle={16}
       contentOffset={initialOffset} // iOS only
       onLayout={scrollToInitial} // Android only
+      onMomentumScrollEnd={() => {}} // workaround for useAnimatedScrollHandler.onMomentumEnd not being called (https://github.com/software-mansion/react-native-reanimated/issues/2735)
```
Comment (lidord-wix) [inline src/components/tabController/PageCarousel.tsx:104]: It'll override the `onMomentumScrollEnd` prop that the user might send, no?

**Review by lidord-wix — COMMENTED** [review]
@M-i-k-e-l 
I wrote small comment

File: src/components/tabController/PageCarousel.tsx (hunk 1 above)
Comment (M-i-k-e-l) [reply src/components/tabController/PageCarousel.tsx]: This is VERY strange, if the user sends `onMomentumScrollEnd` the workaround stops working (even though this is overriding it).

**Review by lidord-wix — APPROVED** [review]

File: src/components/tabController/PageCarousel.tsx (hunk 1 above)
Comment (M-i-k-e-l) [reply src/components/tabController/PageCarousel.tsx]: Fixed by adding `useCallback`

**Pull request merged**